### PR TITLE
Added documentation for flattening migrations

### PR DESCRIPTION
**Pull request by Stefanie899**
Added documentation for flattening migrations, as this is something that is rarely done but extremely helpful for larger project.

Testing this would be cumbersome with little benefit in my opinion (the important portion of code, getting the index of the flattened migration in the overall list of migrations, does not lend itself well to testing), so I opted not to write automated tests. Therefore, automated test coverage *may* decrease from this PR, but it should be minor.

- [N/A] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [ ] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/FlattenedMigration.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                if (this._context == null)
+                {
+                    this._context = new GBApiContext();
+                }
+
+                return _context;
+            }
+        }
+
+        public bool ValidateFlattenedShouldRun(string id)
```
Comment (brandongregoryscott) [inline dotnet/api/Infrastructure/Data.SqlServer/Migrations/FlattenedMigration.cs:44]: Nitpick - alphabetization of methods

File: dotnet/api/Infrastructure/Data.SqlServer/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    * In order to do this, open the flattened migration and ensure it extends `FlattenedMigration` instead of `Migration`.
+    * In the `FlattenedMigration` class, add a new const for the ID of this migration. It should, when sorted as strings, be next in line from the previous flattened migration. Additionally, add it to the static list that contains all the migration ID consts.
+    * Open the `Designer.cs` file of the flattened migration, and update the name in the `[Migration("")]` attribute to use the const you just added.
+    * Rename the files themselves to match what you put in the `[Migration("")]` attribute.
+7. Manually update the flattened migration to call `ValidateFlattenedShouldRun` passing in the ID of the migration you are working off of. This will check if, based on the sort order of the other flattened migrations, this flattened migration has not run on a fresh install. If that method call returns false, you should abort running the flattened migration, as it means this is not a fresh install.
```
Comment (brandongregoryscott) [inline dotnet/api/Infrastructure/Data.SqlServer/README.md:14]: Can some subset of this comment be added as a comment above the method in that base class?

**Review by brandongregoryscott — APPROVED** [review]
Aside from the comment thought, looks good to me 👍

**Comment by wintondeshong** [comment]
@Stefanie899 sorry, i just noticed i didn't "submit" my review so my comments from early this morning were sitting there pending.

File: dotnet/api/Infrastructure/Data.SqlServer/Migrations/FlattenedMigration.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.EntityFrameworkCore;
```
[the remark this answers is not in the document]
Comment (Stefanie899) [reply dotnet/api/Infrastructure/Data.SqlServer/Migrations/FlattenedMigration.cs:1]: I should be able to migrate this over on Friday. Once I do that, I'll also update this repo to use that package (I'll also address the `master` vs `main` issue when I migrate this over).

**Comment by wintondeshong** [comment]
@all-contributors please add @Stefanie899 for code, reviews and documentation

**Pull request merged**